code/API_definitions/qod-api.yaml

```diff
@@ -452,43 +437,112 @@ components:
       content:
         application/json:
           schema:
-            $ref: "#/components/schemas/ErrorInfo"
+            allOf: 
+              - $ref: "#/components/schemas/ErrorInfo"
+              - type: object
+                properties: 
+                  status:
+                    type: string
+                    enum: ["400"]
+                  code:
+                    type: string
+                    enum: ["INVALID_INPUT"]
           example:
+            status: "400"
             code: INVALID_INPUT
             message: "Schema validation failed at  ..."
     Generic401:
       description: Unauthorized
       content:
         application/json:
           schema:
-            $ref: "#/components/schemas/ErrorInfo"
+            allOf: 
+              - $ref: "#/components/schemas/ErrorInfo"
+              - type: object
+                properties: 
+                  status:
+                    type: string
+                    enum: ["401"]
+                  code:
+                    type: string
+                    enum: ["UNAUTHORIZED"]
           example:
+            status: "401"
             code: UNAUTHORIZED
             message: "Authorization failed: ..."
     Generic403:
       description: Forbidden
       content:
         application/json:
           schema:
-            $ref: "#/components/schemas/ErrorInfo"
+            allOf: 
+              - $ref: "#/components/schemas/ErrorInfo"
+              - type: object
+                properties: 
+                  status:
+                    type: string
+                    enum: ["403"]
+                  code:
+                    type: string
+                    enum: ["FORBIDDEN"]
           example:
+            status: "403"
             code: FORBIDDEN
             message: "Operation not allowed: ..."
     SessionNotFound404:
```

Comment:
I could have named it Generic404 as well, but as the only case in this API for 404 was Session Not Found, I chose a more specific name and added an explicit message in the example. Problem with Generic responses is that example has to be generic as well. For automatic API publishing in a web portal it is convenient to embed good examples in the spec, but for client generation purposes it may complicate things. 

I can change the name to `Generic404` if that is more coherent with the rest of responses